extension/src/state_aggregate/accessors.rs

```diff
@@ -0,0 +1,495 @@
+use crate::{
+    accessors::{
+        AccessorIntoIntValues, AccessorIntoValues, AccessorStateIntTimeline, AccessorStateTimeline,
+    },
+    datum_utils::interval_to_ms,
+    raw::{Interval, TimestampTz},
+    state_aggregate::*,
+};
+
+#[pg_operator(immutable, parallel_safe)]
+#[opname(->)]
+pub fn arrow_state_agg_into_values<'a>(
+    agg: StateAgg<'a>,
+    _accessor: AccessorIntoValues<'a>,
+) -> TableIterator<
+    'a,
+    (
+        pgx::name!(state, String),
+        pgx::name!(duration, crate::raw::Interval),
+    ),
+> {
+    into_values_tl(agg)
+}
+#[pg_operator(immutable, parallel_safe)]
+#[opname(->)]
+pub fn arrow_state_agg_into_int_values<'a>(
+    agg: StateAgg<'a>,
+    _accessor: AccessorIntoIntValues<'a>,
+) -> TableIterator<
+    'a,
+    (
+        pgx::name!(state, i64),
+        pgx::name!(duration, crate::raw::Interval),
+    ),
+> {
+    into_values_tl_int(agg)
+}
+
+#[pg_operator(immutable, parallel_safe)]
+#[opname(->)]
+pub fn arrow_state_agg_state_timeline<'a>(
+    agg: StateAgg<'a>,
+    _accessor: AccessorStateTimeline<'a>,
+) -> TableIterator<
+    'a,
+    (
+        pgx::name!(state, String),
+        pgx::name!(start_time, TimestampTz),
+        pgx::name!(end_time, TimestampTz),
+    ),
+> {
+    state_timeline(agg)
+}
+#[pg_operator(immutable, parallel_safe)]
+#[opname(->)]
+pub fn arrow_state_agg_state_int_timeline<'a>(
+    agg: StateAgg<'a>,
+    _accessor: AccessorStateIntTimeline<'a>,
+) -> TableIterator<
+    'a,
+    (
+        pgx::name!(state, i64),
+        pgx::name!(start_time, TimestampTz),
+        pgx::name!(end_time, TimestampTz),
+    ),
+> {
+    state_int_timeline(agg)
+}
+
+macro_rules! interpolated_state_timeline_accessor {
```

Comment:
I don't feel that these macros ended up being very effective here.  You define several macros that you only end up using twice each.  This ends up being much harder to read through and follow, but not that much more efficient at removing replicated code.  The rule of thumb I've always heard is that it's not worth refactoring code until you're using it three times.